include/sundials/sundials_memory.h

```diff
@@ -65,6 +66,10 @@ struct _SUNMemoryHelper
 {
   void*               content;
   SUNMemoryHelper_Ops ops;
+  unsigned long long  num_allocations;
+  unsigned long long  num_deallocations;
```

Comment:
If the memory type is added to `GetAllocStats` (see my comment below) then this could hold the stats for all supported types and the helpers that don't have a content structure currently would not need to add one (or a clone function).

Reply:
I am thinking it would probably make more sense to have GetAllocStats to be an op, still have the content structures, and remove these from the main struct.  Otherwise a helper like Sys has all these extras in the struct that are unnecessary.